module/support/logging/DataLogging/src/DataLogging.cpp

```diff
@@ -158,6 +159,23 @@ namespace module::support::logging {
                     handles = std::move(new_handles);
                 }
             });
+
+        // This checks that we haven't reached the max_size
+        log_check_handler = on<Every<5, std::chrono::seconds>, Sync<DataLog>, Single>().then([this]() {
+            long size = 0;
+            for (auto& f : std::filesystem::recursive_directory_iterator(config.output.directory)) {
+                if (f.is_regular_file()) {
+                    size += f.file_size();
+                }
+            }
+            // If we have past the amount of logging we want to do, disable the logging reaction
+            if (size >= config.output.max_size) {
+                logging_reaction.unbind();
+                log_check_handler.unbind();
```

Comment:
Disabling is probably fine for this
```suggestion
                logging_reaction.disable();
                log_check_handler.disable();
```